bin/check-es-version.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+#shellcheck disable=SC2086
+set -o errexit
+set -o nounset
+set -o pipefail
+
+DATABASE_URL="$1"
+
+function wait_for_request {
+  for _ in {1..10}; do
+    if curl "$DATABASE_URL" > /dev/null 2>&1; then
+      echo "response recieved from $DATABASE_URL"
+      return 0
+    else
+      sleep 1
+    fi
+  done
+
+  echo "Unable to reach Elasticsearch server, please check DATABASE_URL and your database server."
+  return 1
+}
+
+
+#Check that we're using the right Kibana version for the ES we're connecting to
+
+KIBANA_VERSION_PARSER="
+require 'json'
+es_version = JSON.parse(STDIN.read)['version']['number']
+print 4.1 if es_version.start_with?('1.')
+print 4.4 if es_version.start_with?('2.')
+print 5.0 if es_version.start_with?('5.0.')
+print 5.1 if es_version.start_with?('5.1.')
+print 6.0 if es_version.start_with?('6.0.')
+print 6.1 if es_version.start_with?('6.1.')"
+
+wait_for_request
+
+sleep 1
```

Comment:
The sleep "fixes" a race condition in testing, where the `wait_for_request` curl was satisfied by the `echo | nc` loop, but the following `curl` where we try to grab the response body would fail.

```
( while echo '{"version": {"number": "0.8"}}' | nc -l 456; do : ; done ) &
/bin/bash check-es-version.sh http://localhost:456
```

If we should use more reliable way to simulate/provide a response, I'd be happy to implement it.